docs/bnb_smart_chain/staking/index.md

```diff
@@ -0,0 +1,191 @@
+* Validator set are elected in and out based on a staking based governance.
+
+To determinate which validators are eligible to produce blocks, that's why staking mechanism is needed.
+
+### Validator Set
+
+The validator set is the group of nodes that are responsible for validating transactions and producing blocks on the
+BSC. The validator set is determined by the amount of staking credit each validator has, which reflects the amount of
+BNB staked by the validator and its delegators. The top validators with the most staking credit are selected as the
+active validator set, and they take turns to propose and vote on blocks. The rest of the validators are in the standby
+validator set, and they can join the active validator set if their staking credit increases or if some active validators
+drop out.
+
+Validators can join or leave the validator set by creating or editing their validator information.
+Validators can also be removed from the validator set by slashing, which is a penalty for misbehaving or being offline.
+
+### Validator Election
+
+There are different roles for validators:
+
+* Cabinet: the top K (which is 21 currently) validators who get the most chance of producing blocks.
+* Candidate: the top (K, K+NumOfCandidates] validators who get a small chance of producing blocks.
+* Inactive: the reset validators who get no chance of producing blocks.
+
+![img](../img/staking/validator-election.png)
+
+The validator set roles are determined every 24 hours based on the latest staking information.
+
+After UTC 00:00, the consensus engine sorts validators and updates the BSC validator set with the ranking information.
+
+### System Contracts
+
+There are several built-in contracts (i.e., system contracts) to facilitate the BSC staking.
+
+* Validator Set Contract. The contract periodically elects a validator set.
+  The contract also serves as a vault for temporarily storing validator rewards.
+
+* System Reward Contract. This contract acts as a vault to collect part of transaction fees. The funds are used for
+  various public purposes, like distributing fast finality rewards.
+
+* Slash Contract. This contract is used to keep track of the number of times a validator becomes unavailable and
+  triggers penalties once a certain threshold is reached. Additionally, this contract also handles other types of slash
+  events, such as double signing and malicious voting in fast finality.
+
+* Stake Hub Contract. This contract serves as the entry point for managing validators and delegations,
+  while also implementing the logic for slashing specific validators. For delegation/undelegation/redelegation
+  operations, it will call different validators' implementation contracts to manage a user's stake.
+
+### Credit Contract
+
+Each validator has its own validator contract that manages staking credit and facilitates the exchange
+between credit and BNB. The token name of a staking credit is "stake {{validator moniker}} credit",
```

Comment:
+ "Whenever a user delegates BNB, an equivalent quantity of credit tokens are created. On the other hand, when a user withdraws their delegation, a corresponding amount of credit tokens are destroyed, thereby releasing the BNB."

Reply:
added